### PR TITLE
[feat] more info about prerendering errors and new strict option

### PR DESCRIPTION
**Pull request by dummdidumm**
Helps with #7183 and #7244

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: documentation/docs/13-page-options.md [hunk 1]
```diff
@@ -72,13 +72,15 @@ Note that this will disable client-side routing for any navigation from this pag
...
 
 Since these routes cannot be dynamically server-rendered, this will cause errors when people try to access the route in question. There are two ways to fix it:
 
-* Ensure that SvelteKit can find the route by following links from [`config.kit.prerender.entries`](/docs/configuration#prerender). The pages containing the links (e.g. your `/` page) must _themselves_ be prerenderable, or they will be ignored
+* Ensure that SvelteKit can find the route by following links from [`config.kit.prerender.entries`](/docs/configuration#prerender). Add links to dynamic routes (i.e. pages with `[parameters]` ) to this option if they are not found through crawling the other entry points, else they are not prerendered because SvelteKit doesn't know what value the parameters should have. The pages containing the links (e.g. your `/` page) must _themselves_ be prerenderable, or they will be ignored.
```
Comment (dummdidumm) [inline documentation/docs/13-page-options.md:79]: @Rich-Harris could you clarify what you mean with the last sentence?

File: documentation/docs/13-page-options.md [hunk 2]
```diff
@@ -72,13 +72,15 @@ Note that this will disable client-side routing for any navigation from this pag
...
 * Change `export const prerender = true` to `export const prerender = 'auto'`. Routes with `'auto'` can be dynamically server rendered
 
+If you are using `@sveltejs/adapter-static`, _all_ pages (and endpoints, if any) must be prerendered, else you need to use a different adapter.
```
Comment (Rich-Harris) [inline documentation/docs/13-page-options.md:82]:
```suggestion
If you are using `@sveltejs/adapter-static`, _all_ pages (and endpoints, if any) must be prerendered, otherwise you need to use a different adapter.
```

File: packages/kit/src/core/prerender/prerender.js [hunk 3]
```diff
@@ -431,9 +431,9 @@ export async function prerender() {
 
 	if (not_prerendered.length > 0) {
 		throw new Error(
-			`The following routes were marked as prerenderable, but were not prerendered:\n${not_prerendered.map(
+			`The following routes were marked as prerenderable, but were not prerendered, because they were not found while crawling your app:\n${not_prerendered.map(
 				(id) => `  - ${id}`
-			)}\n\nSee https://kit.svelte.dev/docs/page-options#prerender-troubleshooting for more info`
+			)}\n\nSee https://kit.svelte.dev/docs/page-options#prerender-troubleshooting for more info and how to solve this`
```
Comment (Rich-Harris) [inline packages/kit/src/core/prerender/prerender.js:436]:
```suggestion
			)}\n\nSee https://kit.svelte.dev/docs/page-options#prerender-troubleshooting for info on how to solve this`
```

**Review by Rich-Harris — COMMENTED** [review]
a few minor tweaks

File: documentation/docs/13-page-options.md (hunk 1 above)
Comment (Rich-Harris) [reply documentation/docs/13-page-options.md:79]: If you have a prerenderable page like `blog/[slug]`, but the `blog` page that links to individual blog posts is _not_ prerenderable, then the blog posts won't be found

File: documentation/docs/16-configuration.md [hunk 4]
```diff
@@ -252,7 +252,7 @@ See [Prerendering](/docs/page-options#prerender). An object containing zero or m
 - `concurrency` — how many pages can be prerendered simultaneously. JS is single-threaded, but in cases where prerendering performance is network-bound (for example loading content from a remote CMS) this can speed things up by processing other tasks while waiting on the network response
 - `crawl` — determines whether SvelteKit should find pages to prerender by following links from the seed page(s)
 - `enabled` — set to `false` to disable prerendering altogether
-- `entries` — an array of pages to prerender, or start crawling from (if `crawl: true`). The `*` string includes all non-dynamic routes (i.e. pages with no `[parameters]` )
+- `entries` — an array of pages to prerender, or start crawling from (if `crawl: true`). The `*` string includes all non-dynamic routes (i.e. pages with no `[parameters]` ), because SvelteKit doesn't know what value the parameters should have
```
Comment (Rich-Harris) [inline documentation/docs/16-configuration.md:255]:
```suggestion
- `entries` — an array of pages to prerender, or start crawling from (if `crawl: true`). The `*` string includes all non-dynamic routes (i.e. pages with no `[parameters]`, because SvelteKit doesn't know what value the parameters should have)
```

File: documentation/docs/13-page-options.md (hunk 1 above)
Comment (dummdidumm) [reply documentation/docs/13-page-options.md:79]: I see, thanks. Could we rephrase this to something like "Pages not marked as prerenderable will be ignored and their links to other pages will not be crawled"?

File: packages/adapter-static/index.js [hunk 5]
```diff
@@ -26,11 +26,23 @@ export default function (options) {
 
 				if (dynamic_routes.length > 0) {
 					const prefix = path.relative('.', builder.config.kit.files.routes);
+					const config_option =
```
Comment (dummdidumm) [inline packages/adapter-static/index.js:29]: Looking at this again, I think the best way to do this is "either dynamic routes are in the unprerendered pages or the entries option is not the default, then show this message. And if there are dynamic routes then add that sentence in the paranthesis). So basically what I had before but with `||` instead of `&&`

File: packages/kit/src/core/adapt/builder.js [hunk 6]
```diff
@@ -33,10 +33,11 @@ export function create_builder({ config, build_data, routes, prerendered, log })
 		async createEntries(fn) {
 			/** @type {import('types').RouteDefinition[]} */
 			const facades = routes.map((route) => {
+				/** @type {Set<import('types').HttpMethod>} */
 				const methods = new Set();
 
 				if (route.page) {
-					methods.add('SET');
+					methods.add('GET');
```
Comment (Rich-Harris) [inline packages/kit/src/core/adapt/builder.js:40]: lol oops

File: documentation/docs/13-page-options.md [hunk 7]
```diff
@@ -72,13 +72,15 @@ Note that this will disable client-side routing for any navigation from this pag
...
 
-* Ensure that SvelteKit can find the route by following links from [`config.kit.prerender.entries`](/docs/configuration#prerender). The pages containing the links (e.g. your `/` page) must _themselves_ be prerenderable, or they will be ignored
+* Ensure that SvelteKit can find the route by following links from [`config.kit.prerender.entries`](/docs/configuration#prerender). Add links to dynamic routes (i.e. pages with `[parameters]` ) to this option if they are not found through crawling the other entry points, else they are not prerendered because SvelteKit doesn't know what value the parameters should have. Pages not marked as prerenderable will be ignored and their links to other pages will not be crawled, even if some of them would be prerenderable.
 * Change `export const prerender = true` to `export const prerender = 'auto'`. Routes with `'auto'` can be dynamically server rendered
 
+If you are using `@sveltejs/adapter-static`, _all_ pages (and endpoints, if any) must be prerendered, otherwise you need to use a different adapter.
+
```
Comment (Rich-Harris) [inline documentation/docs/13-page-options.md:83]: I think this is no longer true now that we have `strict: false`?

```suggestion
```
